source/blender/blenkernel/intern/python_component.c

```diff
@@ -378,6 +379,25 @@ static void create_properties(PythonComponent *pycomp, PyObject *cls)
         }
       }
     }
+    else if (PyType_Check(pyvalue)) {
+        const char *tp_name = ((PyTypeObject *) pyvalue)->tp_name;
+
+        free = true;
+
+#define PT_DEF(name, lower, upper) \
+        if (!strcmp(tp_name, STRINGIFY(name))) { \
```

Comment:
Not a very robust method to check types. I'd like to use `issubtype` instead, but don't know how to get a reference to each Python type instance here.